src/go/cmd/http-relay-server/server/broker.go

```diff
@@ -280,10 +292,19 @@ func (r *broker) SendResponse(resp *pb.HttpResponse) error {
 }
 
 func (r *broker) ReapInactiveRequests(threshold time.Time) {
-	r.m.Lock()
+	inactiveResps := make([]string, 0)
 	for id, pr := range r.resp {
 		if pr.lastActivity.Before(threshold) {
 			slog.Info("Timeout on inactive request", slog.String("ID", id))
+			pr.markReap <- struct{}{}
+			inactiveResps = append(inactiveResps, id)
+		}
+	}
+
+	r.m.Lock()
+	for _, id := range inactiveResps {
+		pr := r.resp[id]
```

Comment:
What if a r.resp[id] is gone in the meantime. What about using a map for the inactiveResps?

for id, pr := range r.resp {
  if _,found := inactiveResps[id]; !found {
    continue
  }
  ...